salt/utils/http.py

```diff
@@ -144,36 +180,85 @@ def query(url,
             url = test_url
             ret['test'] = True
 
-    result = sess.request(
-        method, url, params=params, data=data
-    )
-    log.debug('Response Status Code: {0}'.format(result.status_code))
-    log.trace('Response Headers: {0}'.format(result.headers))
-    log.trace('Response Text: {0}'.format(result.text))
-    log.trace('Response Cookies: {0}'.format(result.cookies.get_dict()))
+    if requests_lib is True:
+        result = sess.request(
+            method, url, params=params, data=data
+        )
+        result_status_code = result.status_code
+        result_headers = result.headers
+        result_text = result.text
+        result_cookies = result.cookies
+    else:
+        request = urllib2.Request(url)
+
+        if url.startswith('https') or port == 443:
+            if not HAS_MATCHHOSTNAME:
+                log.warn(('match_hostname() not available, SSL hostname '
+                         'checking not available. THIS CONNECTION MAY NOT BE SECURE!'))
+            else:
+                hostname = request.get_host()
+                sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                sock.connect((hostname, 443))
+                sockwrap = ssl.wrap_socket(
+                    sock,
+                    ca_certs='/etc/ssl/certs/ca-certificates.crt',
```

Comment:
This will be the real painful part, finding the cert. certiffi has a good example that we can start with